src/classes/CON_AddToCampaign.cls

```diff
@@ -124,11 +124,14 @@ public class CON_AddToCampaign {
                         }
                     }
                 }
-                if(!match && !isSpecialStatus(cmsCompare)) {
+                //create a campaign member status
+                //if this is a special status, only create a status if we have a setting
+                //create as responded if for the "responded" special status, otherwise default to non-responded
+                if(!match && (!isSpecialStatus(cmsCompare) || getSetting(cmsCompare) != cmsCompare)) {
                     sortOrderMax ++;
                     listStatusForInsert.add(new CampaignMemberStatus(campaignId = cIdCompare,
-                                                                Label = cmsCompare,
-                                                                HasResponded = false,
+                                                                Label = getSetting(cmsCompare),
+                                                                HasResponded = cmsCompare == Label.campaignMemberStatusResponded,
```

Comment:
can't multiple different statuses have responded set true?  if so, then should we be looking at settings for this (or something else) rather than only support the one status we have set in custom labels?